### ISSUE TITLE
[ENHANCEMENT]: Inline images should not insert line breaks

### ISSUE THREAD

**Issue by Kangie** [Zettlr/Zettlr#1156]
## Description
Currently when an image is added via Drag & Drop, it is placed on a new line, even when it is dropped inline with text.

## Proposed Changes
A context-aware Drag & Drop should:
* Insert a newline above and below when the image is dragged onto a blank line that has text immediately above and below.
```
Some Text

![Image](..\Image.jpg)

More Text
```
* Not insert any newlines when an images is dragged onto a line already containing text.
```
Some Text ![Image](..\Image.jpg) More Text
```
* Not insert any newlines when they already exist on **both** sides of the chosen position (I sometimes get the below when replacing existing images)
```
Some Text


![Image](..\Image.jpg)


More Text
```

**Comment by Kangie** [comment]
hey @nathanlesage,

The fixes in 1.7.5 are great. Dragging images inline with text doesn't disrupt your workflow!

There's still an edge case where dragging an image onto a blank line that's surrounded by blank lines inserts an additional line of whitespace above and below the image. 

e.g.: 

```markdown
Some text



Some more text
````

when an image is dragged onto the 'middle' blank line, the output is: 

```markdown
Some text


![image](..\image\path.png)


Some more text
````

I don't know if this is intended behaviour - it's not like MarkDown cares about the extra whitespace, and I certainly appreciate the insertion of whitespace where there is text above and below the image location. Pretty minor, but I thought it worth the mention!

**Comment by nathanlesage** [comment]
Yeah, feel free to observe this, and if it becomes a problem, we can tackle it later for sure!
